packages/eslint-plugin/src/rules/no-unnecessary-type-parameters.ts

```diff
@@ -38,19 +38,33 @@ export default createRule({
       const checker = parserServices.program.getTypeChecker();
       let counts: Map<ts.Identifier, number> | undefined;
 
+      // Get the scope in which the type parameters are declared.
+      const scope = context.sourceCode.getScope(node);
+
       for (const typeParameter of tsNode.typeParameters) {
         const esTypeParameter =
           parserServices.tsNodeToESTreeNodeMap.get<TSESTree.TSTypeParameter>(
             typeParameter,
           );
-        const scope = context.sourceCode.getScope(esTypeParameter);
+
+        const smTypeParameterVariable = nullThrows(
```

Comment:
...what does "sm" stand for here?

[Naming] Maybe, `scopedVariable`? Something like that?

Reply:
meh it's fine :)